src/rp2040.ts

```diff
@@ -344,31 +367,58 @@ export class RP2040 {
   }
 
   step() {
-    this.core.executeInstruction();
+    this.core0.executeInstruction();
+    this.core1.executeInstruction();
   }
 
   execute() {
+    this.core0.stopped = false;
+    this.core1.stopped = false;
+    setTimeout(() => this.execute0(), 0);
+  }
+
+  private execute0() {
+    this.clock.resume();
+    this.executeTimer = null;
+    this.isCore0Running = true;
+    for (let i = 0; i < 1000 && !this.core0.stopped && !this.core0.waiting; i++) {
+      this.core0.executeInstruction();
+    }
+    this.isCore0Running = false;
+    if (this.core1.stopped || this.core1.waiting) {
+      this.executeTimer = setTimeout(() => this.execute0(), 0);
+    } else {
+      this.executeTimer = setTimeout(() => this.execute1(), 0);
+    }
+  }
+
+  private execute1() {
     this.clock.resume();
     this.executeTimer = null;
-    this.stopped = false;
-    for (let i = 0; i < 100000 && !this.stopped && !this.core.waiting; i++) {
-      this.core.executeInstruction();
+    for (let i = 0; i < 1000 && !this.core1.stopped && !this.core1.waiting; i++) {
+      this.core1.executeInstruction();
     }
-    if (!this.stopped) {
-      this.executeTimer = setTimeout(() => this.execute(), 0);
+    if (this.core0.stopped || this.core0.waiting) {
```

Comment:
This will make the emulator execute 1000 instructions from core 0, then 1000 instructions from core 1, etc. - the timing between both cores will be totally off, and depending on the actual instructions executed, one core might run much faster than the other core (as instructions may take more than 1 cycle).

It's probably better to keep one execute() function with something like this:
```
  private execute() {
    ...
    for (let i = 0; i < 1000; i++) {
      this.core0.executeInstruction();
      while(this.core1.cycles < this.core0.cycles) {
        this.core1.executeInstruction();
      }
    }
    ...
  }
```

This is still missing the waiting/stopped logic though.

Reply:
You are right. Had updated.